source/speech/manager.py

```diff
@@ -336,12 +336,15 @@ def _removeCompletedFromQueue(self, index):
 			return False, False
 		for seqIndex, seq in enumerate(self._curPriQueue.pendingSequences):
 			lastCommand = seq[-1] if isinstance(seq, list) else None
-			if isinstance(lastCommand, IndexCommand) and index >= lastCommand.index:
-				endOfUtterance = isinstance(self._curPriQueue.pendingSequences[seqIndex + 1][0], EndUtteranceCommand)
-				if endOfUtterance:
-					# Remove the EndUtteranceCommand as well.
-					seqIndex += 1
-				break # Found it!
+			if isinstance(lastCommand, IndexCommand):
+				if index > lastCommand.index:
+					log.debugWarning("Reached speech index %d, but index %d never handled" % (index, lastCommand.index))
```

Comment:
I'd prefer to use f strings.

```suggestion
					log.debugWarning("Reached speech index {index}, but index {lastCommand.index} never handled")
```

If you want to maintain the type conversion you can do:
```suggestion
					log.debugWarning("Reached speech index {index :d}, but index {lastCommand.index :d} never handled")
```

Reply:
I didn't actually put the f at the start of the string, but I'm sure you get what I mean.

Reply:
Nope, I did not, because I wasn't yet aware of this really funky literal! Also couldn't find it documented that quickly.